server/app/controllers/dev/seeding/DevDatabaseSeedController.java

```diff
@@ -93,11 +110,34 @@ public Result clear() {
     if (!isDevOrStaging) {
       return notFound();
     }
+    if (settingsManifest.getVersionCacheEnabled()) {
+      clearCache();
+    }
     resetTables();
     return redirect(routes.DevDatabaseSeedController.index().url())
         .flashing("success", "The database has been cleared");
   }
 
+  /** Remove all content from the cache. */
+  public void clearCache() {
+    if (!isDevOrStaging) return;
+    Version activeVersion = versionRepository.getActiveVersion();
+    clearVersionCache(programsByVersionCache, activeVersion);
+    clearVersionCache(questionsByVersionCache, activeVersion);
+  }
+
+  /**
+   * Remove all content from the version cache, using the active version as the maximum key to
+   * clear.
+   */
+  private void clearVersionCache(SyncCacheApi cache, Version activeVersion) {
+    for (int num = 1; num <= activeVersion.id; num++) {
+      if (cache.get(String.valueOf(num)).isPresent()) {
+        cache.remove(String.valueOf(num));
+      }
+    }
+  }
+
```

Comment:
**suggestion (blocking):** please move this logic to an existing or new class focused on encapsulating caching behavior

Reply:
I realized I can use the AsyncCacheApi to do this and I don't have to change other references, so updated to do that!